### PR TITLE
Pass AdditionalTestArguments to tests

### LINKED ISSUE #23080: UseProjectReference build configuration didn't catch a build error
The change at https://github.com/Azure/azure-sdk-for-net/commit/c60c872828c71c6b4baeaac69ff7b27caa709021 added the `ClientOptions.Default` member that conflicts with `TableClientOptions.Default`.

But the original build passed:
https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1026038&view=logs&j=3dc8fd7e-4368-5a92-293e-d53cefc8c4b3

And only failed in the PackageReference update PR:

https://dev.azure.com/azure-sdk/public/_build/results?buildId=1030732&view=logs&jobId=bafe3a64-ca18-5d04-c5b9-683b89f12bfb

### PR DESCRIPTION
**Pull request by pakrym**
Fixes: https://github.com/Azure/azure-sdk-for-net/issues/23080

### CODE REVIEW

File: eng/pipelines/templates/stages/platform-matrix.json [hunk 1]
```diff
@@ -37,13 +37,11 @@
           }
       },
       "TestMode": {
-          "netcoreapp21_Coverage_Record": {
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/platform-matrix.json:40]: These changes look unrelated to the issue you are fixing. Is there a reason you decided to remove the net5.0 configuration?

File: eng/pipelines/templates/jobs/ci.tests.yml [hunk 2]
```diff
@@ -56,8 +59,9 @@ jobs:
           /p:ServiceDirectory=${{ parameters.ServiceToTest }}
           /p:IncludeSrc=false /p:IncludeSamples=false /p:IncludePerf=false /p:IncludeStress=false
           /p:RunApiCompat=false /p:InheritDocEnabled=false
-          /p:Configuration=$(BuildConfiguration) $(ConvertToProjectReferenceOption)
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/ci.tests.yml:59]: We should probable remove https://github.com/Azure/azure-sdk-for-net/blob/main/eng/pipelines/templates/variables/globals.yml#L4 as well as I don't see any other usages of this variable anymore.

File: eng/pipelines/templates/stages/platform-matrix.json (hunk 1 above)
Comment (pakrym) [reply eng/pipelines/templates/stages/platform-matrix.json:40]: It is somewhat related. This configuration wasn't passing any value of `AdditionalTestArguments` but we already have net5.0 in other generated configurations. So instead of fixing it I've removed it.

File: eng/pipelines/templates/stages/platform-matrix.json [hunk 3]
```diff
@@ -37,13 +37,11 @@
...
+          "netcoreapp31_Coverage_Record": {
               "SupportsRecording": true,
               "CollectCoverage": true,
-              "TestTargetFramework": "netcoreapp3.1"
-          },
-          "net5.0": {
-              "TestTargetFramework": "net5.0"
+              "TestTargetFramework": "netcoreapp3.1",
+              "AdditionalTestArguments": "/p:UseProjectReferenceToAzureClients=false"
```
Comment (weshaggard) [inline eng/pipelines/templates/stages/platform-matrix.json:44]: Isn't this the default value? Do we need to explicitly pass it in this extra configuration?

File: eng/pipelines/templates/jobs/ci.tests.yml [hunk 4]
```diff
@@ -24,6 +24,9 @@ parameters:
 - name: CloudConfig
   type: object
   default: {}
+- name: AdditionalTestArguments
```
Comment (weshaggard) [inline eng/pipelines/templates/jobs/ci.tests.yml:27]: Where is this parameter actually being used? The only usage I see below is variable access and not template parameter access.

Comment (weshaggard) [reply eng/pipelines/templates/jobs/ci.tests.yml:27]: I suspect you just want to set the default value to empty string in globals.yml like we did with the old parameter.

File: eng/pipelines/templates/stages/platform-matrix.json (hunk 3 above)
Comment (pakrym) [reply eng/pipelines/templates/stages/platform-matrix.json:44]: Didn't work without it the `$(AdditionalTestArguments)` literal value was passed to dotnet test.

**Review by maririos — APPROVED** [review]

**Review by weshaggard — APPROVED** [review]

**Comment by pakrym** [comment]
MasOS issues known
